transport/api/v1/handler.go

```diff
@@ -33,6 +33,7 @@ func (h *Handler) initRouter() {
 		r.Post("/register", h.Register)
 		r.Get("/verify", h.Verify)
 		r.Post("/login", h.Login)
+		r.With(authenticate).Post("/refresh", h.Refresh)
```

Comment:
Jestli spravně tomu rozumím, potřebujeme ten refresh v tom připadě kdy máme expired access token. Ale tady máme With(authenticate), a tedy to nebude fungovat jestli máme expired access token a nedostaneme nový access token

Reply:
Jako chápu tvoje obavy, ale imo ten refresh budem volat při nějakejch significant operacích, nebo když se ten token bude blížit expiraci, i ten refresh token má svojí dobu validity a nejsem si jistej, jak je dlouhá

Reply:
Tak podle dokumentace je refresh relativně long-lived oproti access tokenu, takže by to asi šlo bez authentikace, ale pak by se musel promyslet celej ten workflow, jak přijdeme na to, že je token expired a získáme ten refresh token - ten se normálně neposílá s requestama

Reply:
Myslim si, ze workflow je takovy:
1. Behem loginu editor dostava access a refresh token, to uklada nekde
2. Pouziva pri praci s endpointamy jenom access token
3. V ten moment, kdy access token je expired, muze zavolat refresh endpoint (bez authentikace) s refresh tokenem v body requestu a dostava novy access token

Reply:
> V ten moment, kdy access token je expired, muze zavolat refresh endpoint

No ale proč by ho volal? Proč editora s timhle otravovat? Neni lepší, když to budem dělat automaticky my, a tim pádem to můžem dělat i třeba jenom minutu před tim, než vyprchá. Ale zrovna tohle je mi reálně asi jedno

Reply:
při volaní s refresh tokenem se vytvoří nový access token. Jestli budeme to volat my, jak editor dostane nový access token? Nedavá mi to smysl. Zkus to prosím, možná máme různé rozumění toho

Reply:
Ale vždyť editor jako uživatel o žádnym tokenu ani vědět nebude, ten se posílá v headeru a bude uloženej v local storage prohlížeče
S tokenem bude pracovat akorát tak FE

Reply:
Access token má expiration time 1 hour. Když máme web aplikaci a user nepouživá naš service více než 1 hodinu, nebude mít možnost refreshnout access token, proto potřebujeme refreshovat access token jenom s refresh tokenem a bez access tokenu.

Reply:
No dobře, ale tohle nemusí bejt nechtěný chování. Proč bys chtěl držet uživatele přihlášenýho dýl jak hodinu v aplikaci, kde celý co dělá je že dá (un)subscribe k newsletteru, nebo nějakej newsletter publikuje? To nebude práce na hodiny. Naopak když tomu uživateli necháš možnost pokračovat v session i o několik hodin později, tak se vystavuješ riziku session hijackingu (ne teda že by na tom u nás nějak záleželo). Za mě je naprosto v pohodě, že pokud uživatel se stránkou dýl jak hodinu neinteraguje, tak ho to odhlásí. 
Navíc jak řikám, ty refreshe stejně budeme posílat my za uživatele, tak to rovnou můžeme dělat těsně před expirací, abychom si odpustili problémy s vyhodnocováním, jak dlouho po expiraci to ještě dovolovat, případně při každym requestu posílat header jak s access tokenem, tak refresh tokenem.